src/as_execution/abi.rs

```diff
@@ -930,6 +930,23 @@ where
     Ok(buffer)
 }
 
+/// performs a sha256 hash on bytes and returns the hash as bytes
+#[named]
+pub(crate) fn assembly_script_sha256(mut ctx: FunctionEnvMut<ASEnv>, bytes: i32) -> ABIResult<i32> {
+    let env = ctx.data().clone();
+    sub_remaining_gas_abi(&env, &mut ctx, function_name!())?;
+    let memory = get_memory!(env);
+    let bytes = read_buffer(memory, &ctx, bytes)?;
```

Comment:
I wondering if the gas cost should be dependent of the bytes len we hash?

Reply:
We should study this together and find the right formula

Please see [Gas cost of a sha256 hash](https://ethereum.stackexchange.com/questions/76110/gas-cost-of-a-sha256-hash)

Reply:
Actually, we don't have a dynamic gas cost because parameter size in other ABI doesn't make a lot of diff. But it makes sens in a hashing function. It'll be added in the future

Reply:
done

Reply:
Instead of a TODO, it's better to open an issue no?

Reply:
It makes sense. See https://github.com/massalabs/massa-sc-runtime/issues/228